src/models/checkList/base.check.js

```diff
@@ -62,6 +42,14 @@ module.exports = class BaseCheck {
     }
   }
 
+  async getUploadedFileDetails (fileSubject, propertyName) {
+    const fileDetails = this.data[propertyName]
+    if (!fileDetails) {
+      this.data[propertyName] = await Annotation.listByApplicationIdAndSubject(this.data, fileSubject)
```

Comment:
As we discussed, `propertyName` seems to be redundant here.  Currently it is just used to ensure that the properties on the `data` object are unchanged by this refactor, but as those properties don't appear to be accessed anywhere else then that shouldn't matter.
However, we agreed to treat this as a separate refactoring exercise for another PR.